warehouse/integrations/clickhouse/clickhouse_test.go

```diff
@@ -36,210 +43,277 @@ import (
 
-	var (
-		provider = warehouseutils.CLICKHOUSE
-		jobsDB   = testhelper.SetUpJobsDB(t)
-		tables   = []string{"identifies", "users", "tracks", "product_track", "pages", "screens", "aliases", "groups"}
-	)
+	jobsDBPort := c.Port("jobsDb", 5432)
+	minioPort := c.Port("minio", 9000)
+	transformerPort := c.Port("transformer", 9090)
+	port := c.Port("clickhouse", 9000)
+	clusterPort1 := c.Port("clickhouse01", 9000)
+	clusterPort2 := c.Port("clickhouse02", 9000)
+	clusterPort3 := c.Port("clickhouse03", 9000)
+	clusterPort4 := c.Port("clickhouse04", 9000)
 
-	testCases := []struct {
-		name                        string
-		sourceID                    string
-		destinationID               string
-		writeKey                    string
-		warehouseEvents             testhelper.EventsCountMap
-		warehouseModifiedEvents     testhelper.EventsCountMap
-		clusterSetup                func(t testing.TB)
-		db                          *sql.DB
-		s3EngineEnabledWorkspaceIDs []string
-	}{
-		{
-			name:          "Single Setup",
-			sourceID:      "1wRvLmEnMOOxNM79pwaZhyCqXRE",
-			destinationID: "21Ev6TI6emCFDKph2Zn6XfTP7PI",
-			writeKey:      "C5AWX39IVUWSP2NcHciWvqZTa2N",
-			db:            dbs[0],
-		},
-		{
-			name:          "Single Setup with S3 Engine",
-			sourceID:      "1wRvLmEnMOOxNM79pwaZhyCqXRE",
-			destinationID: "21Ev6TI6emCFDKph2Zn6XfTP7PI",
-			writeKey:      "C5AWX39IVUWSP2NcHciWvqZTa2N",
-			db:            dbs[0],
-			s3EngineEnabledWorkspaceIDs: []string{
-				"BpLnfgDsc2WD8F2qNfHK5a84jjJ",
-			},
-		},
-		{
-			name:          "Cluster Mode Setup",
-			sourceID:      "1wRvLmEnMOOxNM79ghdZhyCqXRE",
-			destinationID: "21Ev6TI6emCFDKhp2Zn6XfTP7PI",
-			writeKey:      "95RxRTZHWUsaD6HEdz0ThbXfQ6p",
-			db:            dbs[1],
-			warehouseModifiedEvents: testhelper.EventsCountMap{
-				"identifies":    8,
-				"users":         2,
-				"tracks":        8,
-				"product_track": 8,
-				"pages":         8,
-				"screens":       8,
-				"aliases":       8,
-				"groups":        8,
-			},
-			clusterSetup: func(t testing.TB) {
-				t.Helper()
-				initializeClickhouseClusterMode(t, dbs[1:], tables)
-			},
-		},
-		{
-			name:          "Cluster Mode Setup with S3 Engine",
-			sourceID:      "1wRvLmEnMOOxNM79ghdZhyCqXRE",
-			destinationID: "21Ev6TI6emCFDKhp2Zn6XfTP7PI",
-			writeKey:      "95RxRTZHWUsaD6HEdz0ThbXfQ6p",
-			db:            dbs[1],
-			warehouseEvents: testhelper.EventsCountMap{
-				"identifies":    8,
-				"users":         2,
-				"tracks":        8,
-				"product_track": 8,
-				"pages":         8,
-				"screens":       8,
-				"aliases":       8,
-				"groups":        8,
-			},
-			warehouseModifiedEvents: testhelper.EventsCountMap{
-				"identifies":    8,
-				"users":         2,
-				"tracks":        8,
-				"product_track": 8,
-				"pages":         8,
-				"screens":       8,
-				"aliases":       8,
-				"groups":        8,
-			},
-			s3EngineEnabledWorkspaceIDs: []string{
-				"BpLnfgDsc2WD8F2qNfHK5a84jjJ",
-			},
-		},
```

Comment:
This has been removed since now we are running the rudder server locally and accessing the minio and clickhouse using localhost, it is not possible to use the same host&port configured for load files generation and [loadByCopyCommand](https://github.com/rudderlabs/rudder-server/blob/e0280e4bf35323844fc0436a56f4803a739b8cd7/warehouse/integrations/clickhouse/clickhouse.go#L531) because clickhouse can't access minio using the same host&port.

Reply:
This has already been covered in a unit test. https://github.com/rudderlabs/rudder-server/blob/e0280e4bf35323844fc0436a56f4803a739b8cd7/warehouse/integrations/clickhouse/clickhouse_test.go#L341